src/groups/mqb/mqbblp/mqbblp_storagemanager.cpp

```diff
@@ -742,10 +745,15 @@ void StorageManager::processPartitionSyncEvent(
         return;  // RETURN
     }
 
+    PartitionInfo                    pinfo;
```

Comment:
Could you explain this change to me?  Is the worry that `clusterState` will be mutated?  And if so, could it be mutated while we're constructing this `PartitionInfo`?

Reply:
There is another call to `mqbc::StorageUtil::validatePartitionSyncEvent()` from [`mqbc::StorageManager`](https://github.com/kaikulimu/blazingmq/blob/f3d03d74c4d6bb335080b1d4120d58bcaebe5fc3/src/groups/mqb/mqbc/mqbc_storagemanager.cpp#L2755-L2763). That call is from the *QUEUE DISPATCHER* thread, and thus is not safe to access `clusterState`. Hence, I changed the signature of `validatePartitionSyncEvent()` to take in a `PartitionInfo` instead of a `ClusterState`.

The change you see here is simply a bi-product. There is no worry for mutation.

Reply:
lgtm then